pandas/core/algorithms.py

```diff
@@ -981,14 +981,60 @@ def select_n_frame(frame, columns, n, method, keep):
     -------
     nordered : DataFrame
     """
-    from pandas.core.series import Series
+    from pandas import Int64Index
     if not is_list_like(columns):
         columns = [columns]
     columns = list(columns)
```

Comment:
Do u want to reference the column in the exception? IOW 
```python
def _ensure_is_valid_dtype_n_method(method, dtype):
    if not ((is_numeric_dtype(dtype) and not is_complex_dtype(dtype)) or
                needs_i8_conversion(dtype)):
            raise TypeError("Cannot use method '{method}' with "
                            "dtype {dtype}".format(method=method, dtype=dtype))
```
or something like

```python
def _is_valid_dtype_n_method(dtype):
    return ((is_numeric_dtype(dtype) and not is_complex_dtype(dtype)) or
                needs_i8_conversion(dtype)):
```
and then in select_n_frame function something like
```python
if not _is_valid_dtype_n_method(dtype):
    raise TypeError("Column `column` has dtype `dtype` can't use n with column")
```
and then in select_n_series function something like
```python
if not _is_valid_dtype_n_method(dtype):
    raise TypeError("Cannot use  method `method` with dtype")
```